src/backend/catalog/namespace.c

```diff
@@ -953,33 +954,50 @@ FuncnameGetCandidates(List *names, int nargs, List *argnames,
 {
 	FuncCandidateList resultList = NULL;
 	bool		any_special = false;
-	char	   *schemaname;
+	char	   *schemaname = NULL;
```

Comment:
Without setting schemaname and modulename to NULL at initialization, I was getting a segfault when trying to install postgres. I will have to work with the compiler warnings to find all the spots to fix that need to explicitly be set to NULL instead of whatever else it points to otherwise. The premature initialization was my way of setting to NULL in one place and overwriting that with some valid string in the necessary places, instead of figuring out all the places where I had to set it to NULL so any garbage value won't parse as a string. I will spend some time looking at where all those are, having taken a shortcut in the first pass to get to some code that works.

Reply:
I'm pushing updates for the other comments, and will come back to figuring out whats going on with the seg faults without default initialization in a separate commit all its own.

Reply:
Okay, fixed this on latest commit!